content/en/docs/setup/install/virtual-machine/index.md

```diff
@@ -32,64 +27,33 @@ but not production. Like all alpha features, this guide is subject to change.
+1. Create a virtual machine
+1. Set the environment variables `"${VM_NAME}"`, `"${WORK_DIR}"`,`"${CLUSTER_NAME}"` , `"${VM_NAMESPACE}"`,
+，`"${SERVICE_ACCOUNT}"` your cluster name, and the service namespace. Ensure `"${WORK_DIR}"` is prefixed with `"${HOME}"`
     (e.g. `WORK_DIR="${HOME}"/vmintegration`).
 
     {{< text bash >}}
-    $ ISTIO_DIR="<the directory containing an unarchived version of Istio>"
+    $ VM_NAME="<the name of your vm instance you created>"
     $ CLUSTER_NAME="<the name of your cluster>"
-    $ SERVICE_NAMESPACE="<the name of your service namespace>"
+    $ VM_NAMESPACE="<the name of your service namespace>"
     $ WORK_DIR="<a certificate working directory>"
+    $ SERVICE_ACCOUNT="<name of the Kubernetes service account you want to use for your VM>"
     {{< /text >}}
 
-1. Create the `"${WORK_DIR}"/"${CLUSTER_NAME}"/"${SERVICE_NAMESPACE}"` working directories.
-
-    {{< text bash >}}
-    $ mkdir -p "${WORK_DIR}"/"${CLUSTER_NAME}"/"${SERVICE_NAMESPACE}"
-    {{< /text >}}
-
-## Create certificates for use with the virtual machine and Istio control plane
-
-{{< tip >}}
-This `Makefile` is limited to creating one virtual machine certificate per cluster. The Istio authors
-expect operators to read and understand this guide to formulate their own plans for creating and
-managing virtual machines. It is important for you to read and understand this `Makefile` for any
-deployment you place into production.
-{{< /tip >}}
-
-1. Execute the following commands to create certificates for use by Istio. See
-    [Certificate Authority (CA) certificates](/docs/tasks/security/cert-management/plugin-ca-cert/)
-    for more details on configuring an external CA. The `NAME` variable is
-    used during certificate generation to uniquely identify clusters. The
-    `NAMESPACE` variable identifies the namespace where the virtual machine
-    connectivity is hosted.
-
-    {{< text bash >}}
-    $ cd "${WORK_DIR}"
-    $ make -f "${ISTIO_DIR}"/tools/certs/Makefile NAME="${CLUSTER_NAME}" NAMESPACE="${SERVICE_NAMESPACE}" "${CLUSTER_NAME}"-cacerts-selfSigned
-    {{< /text >}}
-
-1. Execute the following commands to create certificates for use on the virtual machine.
+1. Create the `"${WORK_DIR}"/"${CLUSTER_NAME}"/"${VM_NAMESPACE}"` working directories.
 
     {{< text bash >}}
-    $ cd "${WORK_DIR}"
-    $ make -f "${ISTIO_DIR}"/tools/certs/Makefile NAME="${CLUSTER_NAME}" NAMESPACE="${SERVICE_NAMESPACE}" "${NAMESPACE}"-certs-selfSigned
+    $ mkdir -p "${WORK_DIR}"/"${CLUSTER_NAME}"/"${VM_NAMESPACE}"
     {{< /text >}}
 
 ## Install the Istio control plane
 
-The Istio control plane must be installed with virtual machine integration enabled (`values.global.meshExpansion.enabled: true`).
+ Set the IstioOperator spec `values.global.meshExpansion.enabled: true`
 
-1. Register the certificates needed for installation.
+1. Create namespace to install Istio.
```

Comment:
nit: this is technically not needed

Reply:
yes we need a namespace to store the pod and create a token based on this @howardjohn do this namespace are required for this token ?

Reply:
You need the namespace for the VM - but `istioctl install` will create istio-system for you. It doesn't really matter either way though

Reply:
I know but this step: `echo '{"kind":"TokenRequest","apiVersion":"authentication.k8s.io/v1","spec":{"audiences":["istio-ca"],"expirationSeconds":'$tokenexpiretime'}}' | kubectl create --raw /api/v1/namespaces/$VM_NAMESPACE/serviceaccounts/$SERVICE_ACCOUNT/token -f - | jq -j '.status.token' > "${WORK_DIR}"/"${CLUSTER_NAME}"/"${VM_NAMESPACE}"/istio-token`

needs a vm namespace

Reply:
I now don't need to create it. But in the latter use, the user needs to create it. Ok will remove it

Reply:
`kubectl -n $VM_NAMESPACE get configmaps istio-ca-root-cert -o json | jq -j '."data"."root-cert.pem"' > "${WORK_DIR}"/"${CLUSTER_NAME}"/"${VM_NAMESPACE}"/root-cert`

and you need a namespace to extract the root cert

Reply:
yes delete it